### DISCUSSION TITLE
Android In App Messaging Build Gradle Failing for SDK version

### DISCUSSION THREAD

**Discussion by bpfeiffer187** [invertase/react-native-firebase#4423] [Q&A]
Anyone else run into the issue with adding in app messaging becuase it tries to dynamically pull in SDK version. 

project.ext {
  set('react-native', [
    versions: [
      android : [
        minSdk    : jsonMinSdk,
        targetSdk : jsonTargetSdk,
        compileSdk: jsonCompileSdk,
        // optional as gradle.buildTools comes with one by default
        // overriding here though to match the version RN uses
        buildTools: jsonBuildTools
      ],

i notice other packages specify the versions

project.ext {
  set('react-native', [
    versions: [
      android : [
        minSdk    : 16,
        targetSdk : 28,
        compileSdk: "28.0.3"
        // optional as gradle.buildTools comes with one by default
        // overriding here though to match the version RN uses
        buildTools: jsonBuildTools
      ],

**Comment by mikehardy** [comment]
You appear to be on old versions. *all* the packages should be getting it dynamically

 https://github.com/invertase/react-native-firebase/pull/4122

**Reply by bpfeiffer187** [reply]
when where they updated? i just did the implmentation for core and analytics about 3 weeks ago

**Reply by mikehardy** [reply]
https://rnfirebase.io/releases - the PR is linked above, you can browse the code? I'm just saying I personally did a PR that *should* have had the effect of them *all* being dynamic. So I'm surprised there are any non-dynamic references to the build version and I suspect you have mixed versions or something, but you've provided no information with which to confirm or refute the assertion

**Reply by bpfeiffer187** [reply]
yea i do have mixed becuase i put in app and analytics about a month ago and those 

roject.ext {
  set('react-native', [
    versions: [
      android           : [
        minSdk    : 16,
        targetSdk : 28,
        compileSdk: 28,
        // optional as gradle.buildTools comes with one by default
        // overriding here though to match the version RN uses
        buildTools: "28.0.3"
      ]

messaging looks like

project.ext {
  set('react-native', [
    versions: [
      android : [
        minSdk    : jsonMinSdk,
        targetSdk : jsonTargetSdk,
        compileSdk: jsonCompileSdk,
        // optional as gradle.buildTools comes with one by default
        // overriding here though to match the version RN uses
        buildTools: jsonBuildTools
      ],

      firebase: [
        bom: firebaseBomVersion,
      ],
    ],
  ])
}

so if i update the other packages will i get the same error i am getting with messaging?

where would the SDK versions then be defined?

**Reply by bpfeiffer187** [reply]
also thank you for thee help and convo

**Reply by mikehardy** [reply]
update everything to current stable, it will work - or, it should work...

**Reply by bpfeiffer187** [reply]
yea that looks like the app gradle errors went away. However now when i run it on emulator it crashes right away.

im digging inot that now.

**Reply by mikehardy** [reply]
Ah, but instant android crashes are actually the easiest, those are usually my favorite. `adb logcat` and you get a stack trace right there, no need for a full IDE or anything, and the stack usually has a good error. This confirms my suspicion about the root cause though, you had versions straddling the PR I linked above, and that won't work well, no
